homeassistant/helpers/entity_registry.py

```diff
@@ -50,6 +50,15 @@
 if TYPE_CHECKING:
     from homeassistant.config_entries import ConfigEntry
 
+# pylint: disable-next=fixme
+# TODO 3.9: Replace EntityRegistryUserDict with generic UserDict at runtime
+if TYPE_CHECKING:
+    # pylint: disable-next=unsubscriptable-object
+    EntityRegistryUserDict = UserDict[str, "RegistryEntry"]
+else:
+    EntityRegistryUserDict = UserDict
```

Comment:
Instead of introducing all kinds of workarounds, we can also wait a little bit, as we are close to dropping Python 3.8

Reply:
Didn't know that it was this close 😄
Pushed a new commit to remove the workaround now that 3.8 has been removed.